pkg/agent/agent.go

```diff
@@ -239,19 +252,17 @@ func (a *Agent) Run(ctx context.Context) error {
 		Metrics: metrics,
 	})
 
-	endpoints := a.newEndpoints(metrics, manager, workloadAttestor)
-
-	if err := healthChecker.AddCheck("agent", a); err != nil {
-		return fmt.Errorf("failed adding healthcheck: %w", err)
-	}
+	agentEndpoints := a.newEndpoints(metrics, manager, workloadAttestor)
+	go func() {
+		agentEndpoints.WaitForListening(readyForHealthChecks)
+	}()
 
+	a.started = true
```

Comment:
Could we do this in the go routine, after we've finished waiting for the endpoints to start listening? I think that's kind of the intention with this flag.

Reply:
the one up a couple lines? Yeah, I think that makes sense.